tests/org.eclipse.ui.tests/Eclipse UI Tests/org/eclipse/ui/tests/markers/MarkerTypeTests.java

```diff
@@ -0,0 +1,142 @@
+/*******************************************************************************
+ * Copyright (c) 2022 Enda O'Brien and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which accompanies this distribution,
+ * and is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors: IBM Corporation - initial API and implementation
+ *******************************************************************************/
+
+package org.eclipse.ui.tests.markers;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+import java.lang.reflect.Field;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.HashSet;
+
+import org.eclipse.ui.PlatformUI;
+import org.eclipse.ui.internal.views.markers.ExtendedMarkersView;
+import org.eclipse.ui.internal.views.markers.MarkerContentGenerator;
+import org.eclipse.ui.tests.NoApplicationAttribTestView;
+import org.eclipse.ui.tests.SubTypeOnlyTestView;
+import org.eclipse.ui.tests.TypeAndSubTypeTestView;
+import org.eclipse.ui.tests.TypeOnlyTestView;
+import org.eclipse.ui.views.markers.MarkerSupportView;
+import org.eclipse.ui.views.markers.internal.ContentGeneratorDescriptor;
+import org.eclipse.ui.views.markers.internal.MarkerType;
+import org.eclipse.ui.views.markers.internal.MarkerTypesModel;
+import org.junit.Test;
+
+public class MarkerTypeTests {
```

Comment:
you need to add the Test to a suite, so that it is executed on CI

Reply:
https://ci.eclipse.org/platform/job/eclipse.platform.ui/job/PR-144/15/testReport/UiTestSuite%20org.eclipse.ui.tests.internal.InternalTestSuite%20org.eclipse.ui.tests.markers/MarkerTypeTests/